runtests.py

```diff
@@ -646,21 +647,28 @@ def write(self, value):
 
     def _collect(self):
         s = ''.join(self.output)
-        results = {'errors': [], 'warnings': []}
+        results = {'error': [], 'warning': [], 'performance hint': []}
         for line in s.splitlines():
             match = self.match_error(line)
             if match:
-                is_warning, line, column, message = match.groups()
-                results['warnings' if is_warning else 'errors'].append((int(line), int(column), message.strip()))
+                message_type, line, column, message = match.groups()
+                if not message_type:
+                    message_type = 'error'
+                else:
+                    message_type = message_type[:-1]  # drop final :
+                results[message_type].append((int(line), int(column), message.strip()))
 
-        return [["%d:%d: %s" % values for values in sorted(results[key])] for key in ('errors', 'warnings')]
+        return [["%d:%d: %s" % values for values in sorted(results[key])] for key in ('error', 'warning', 'performance hint')]
 
     def geterrors(self):
         return self._collect()[0]
 
     def getwarnings(self):
         return self._collect()[1]
 
+    def getperformancehints(self):
+        return self._collect()[2]
+
```

Comment:
Is this used anywhere? Do we need to add it? (And, BTW, are the other two corresponding methods really used anywhere?)